include/meta/meta.hpp

```diff
@@ -2125,6 +2125,21 @@ namespace meta
             using min = defer<min, T, U>;
         }
 
+        /// Index of the first occurrence of the type \p T within the list \p List.
+        /// \par Complexity
+        /// \f$ O(N) \f$.
+        /// \ingroup list
+        template <typename T, typename List>
```

Comment:
I don't love returning `nil_`. Since it's not an integral constant wrapper, it could cause errors in compile-time branches that aren't taken but get instantiated anyway. As for returning `List::size()`, there's not much worry about out of bounds indexing at compile-time. It just won't compile. Some argument can be made for returning `npos`, where `npos` is `size_t<-1>`. That would help with pattern matching.

Regarding the trailing underscore for `nil_`, it's because Fusion had to change `fusion::nil` to `fusion::nil_` due to the fact that Apple defines `nil` as a macro. IIRC. Crummy, I know.